### PR TITLE
HOSS2-181 - Handle errors outside of the errorhandler

### PR DESCRIPTION
**Pull request by burns238**

### CODE REVIEW

File: app/controllers/StatusResultController.scala [hunk 1]
```diff
@@ -51,22 +53,28 @@ class StatusResultController @Inject()(
...
-        logger.info(s"Match found with no statuses - CorrelationId: $correlationId")
+    error match {
+      case StatusCheckConflict => Ok(multipleMatchesFoundPage(query))
+      case StatusCheckNotFound => Ok(statusCheckFailurePage(query))
+      case _                   => InternalServerError(externalErrorPage())
+    }
+
+  private def displaySuccessfulResult(query: StatusCheckByNinoFormModel)(response: StatusCheckResponse)(
+    implicit request: Request[AnyContent]): Result =
+    response.result.statuses match {
```
Comment (liampan) [inline app/controllers/StatusResultController.scala:72]: awful

File: app/models/HomeOfficeError.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.http.Status._
+
+sealed trait HomeOfficeError
+
+case object StatusCheckNotFound extends HomeOfficeError
```
Comment (liampan) [inline app/models/HomeOfficeError.scala:23]: can we put all these inside of the companion object to HomeOfficeError so they have the same name space

File: app/models/StatusCheckResponse.scala [hunk 3]
```diff
@@ -22,10 +22,8 @@ case class StatusCheckResponse(
   // Identifier associated with a checks,
   // if x-correlation-id is not provided in request headers, a new id generated using token service
```
Comment (liampan) [inline app/models/StatusCheckResponse.scala:23]: can we clear all these code coments that arent helpful

File: conf/logback.xml [hunk 4]
```diff
@@ -45,7 +45,7 @@
         <appender-ref ref="STDOUT_IGNORE_NETTY" />
     </logger>
 
-    <logger name="application" level="ERROR"/>
+    <logger name="application" level="INFO"/>
```
Comment (liampan) [inline conf/logback.xml:48]: ?

File: it/connectors/HomeOfficeImmigrationStatusConnectorISpec.scala [hunk 5]
```diff
@@ -21,93 +21,68 @@ class HomeOfficeImmigrationStatusConnectorISpec extends HomeOfficeImmigrationSta
...
-        }
+        val result: Either[HomeOfficeError, StatusCheckResponse] = connector.statusPublicFundsByNino(request).futureValue
+
+        result should be ('left)
+        result.left.get shouldBe StatusCheckInternalServerError
       }
     }
   }
 
   val errorGenerator: HttpErrorFunctions = new HttpErrorFunctions {}
```
Comment (liampan) [inline it/connectors/HomeOfficeImmigrationStatusConnectorISpec.scala:84]: is this being used?

File: test/connectors/StatusCheckResponseHttpParserSpec.scala [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    "return a left where a 200 is returned with an invalid json response" in {
+      val responseBody = """{"response": "Something"}"""
+      val response = HttpResponse(OK, responseBody)
+
+      val result: Either[HomeOfficeError, StatusCheckResponse] =
+        StatusCheckResponseReads.read("POST", "some url", response)
+      result.leftValue shouldBe StatusCheckInvalidResponse
+    }
+
+    errors.foreach { case (k, v) => checkError(k, v) }
```
Comment (liampan) [inline test/connectors/StatusCheckResponseHttpParserSpec.scala:73]: k and v have no meaning
either 
```suggestion
    errors.foreach { case (status, error) => checkError(status, error)}
```
or
```suggestion
    errors.foreach { (checkError  _).tupled }
```

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -26,7 +26,8 @@ object AppDependencies {
     "com.github.tomakehurst"      %  "wiremock"                 % "2.27.2",
     "com.vladsch.flexmark"        % "flexmark-all"              % "0.35.10",
     "org.mockito"                 % "mockito-all"               % "1.10.19",
-    "org.pegdown"                 %  "pegdown"                  % "1.6.0"
+    "org.pegdown"                 %  "pegdown"                  % "1.6.0",
+    "com.ironcorelabs"            %% "cats-scalatest"           % "3.1.1",
```
Comment (liampan) [inline project/AppDependencies.scala:30]: is this used now?

**Review by liampan — APPROVED** [review]

File: app/ErrorHandler.scala [hunk 8]
```diff
@@ -58,9 +59,8 @@ class ErrorHandler @Inject()(
     auditServerError(request, exception)
     implicit val r: Request[String] = Request(request, "")
     exception match {
-      case _: NoActiveSession                       => toGGLogin(if (isDevEnv) s"http://${request.host}${request.uri}" else s"${request.uri}")
-      case _: InsufficientEnrolments                => Forbidden
-      case _: HomeOfficeImmigrationStatusProxyError => InternalServerError(externalErrorTemplate())
+      case _: NoActiveSession        => toGGLogin(if (isDevEnv) s"http://${request.host}${request.uri}" else s"${request.uri}")
```
[the remark this answers is not in the document]
Comment (burns238) [reply app/ErrorHandler.scala:62]: I actually think these are fine in here. But yeah if we do anything it'd be a separate task

File: project/AppDependencies.scala (hunk 7 above)
Comment (burns238) [reply project/AppDependencies.scala:30]: Yeah, only in one place currently

File: app/ErrorHandler.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (liampan) [reply app/ErrorHandler.scala:62]: theyre thrown in the AuthAction I think they should be handled there, but sure :)

File: app/ErrorHandler.scala [hunk 9]
```diff
@@ -58,27 +59,18 @@ class ErrorHandler @Inject()(
...
+      case _: InsufficientEnrolments => Forbidden
       case e =>
         logger.error(e.getMessage, e)
-        InternalServerError(internalErrorTemplate())
+        Ok(internalErrorTemplate())
```
Comment (liampan) [inline app/ErrorHandler.scala:66]: should be ISE no?

**Review by liampan — APPROVED** [review]

**Pull request merged**